### PR TITLE
release(upgradetest): confirm postrelease version is newer than the latest sourcegraph version

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Last week, I ran into an obscure error where an internal release build had it's release tests failing. 

[Context](https://sourcegraph.slack.com/archives/C05DWT4ANHH/p1715011866175999)

This led to an hypothesis where we believe the version I was trying to release `5.3.666` was assumed by the tests to be behind the latest version `5.3.12303`. I added this check to ensure this doesn't repeat itself.

## Test plan

Will test during the release testing later.

### CODE REVIEW

File: testing/tools/upgradetest/tests-util.go [hunk 1]
```diff
@@ -800,20 +800,22 @@ func handleVersions(cCtx *cli.Context, overrideStd, overrideMVU, overrideAuto []
...
-		if latestFullVer == nil || tag.GreaterThan(latestFullVer) {
-			latestFullVer = tag
+		if latestFullVer == nil || v.GreaterThan(latestFullVer) {
+			latestFullVer = v
 		}
 		latestMinorVer, err = semver.NewVersion(fmt.Sprintf("%d.%d.0", latestFullVer.Major(), latestFullVer.Minor()))
 		if err != nil {
 			return nil, nil, nil, nil, nil, nil, err
 		}
 	}
```
Comment (BolajiOlajide) [inline testing/tools/upgradetest/tests-util.go:812]: @DaedalusG I took the liberty of tweaking the logic here as I don't think we need the second loop here. Moving this logic after appending to `validTags` will achieve the same result albeit faster.

Let me know if you have concerns

File: testing/tools/upgradetest/tests-util.go [hunk 2]
```diff
@@ -800,20 +800,22 @@ func handleVersions(cCtx *cli.Context, overrideStd, overrideMVU, overrideAuto []
...
 
+	semverPostRelease := semver.MustParse(postRelease)
```
Comment (DaedalusG) [inline testing/tools/upgradetest/tests-util.go:814]: We should probably wrap this in a check to make sure the postRelease flag is set. There are ways to run this test without specifying postRelease targets

Comment (BolajiOlajide) [reply testing/tools/upgradetest/tests-util.go]: Oh I see. Added an `if` check to only do this is the `postRelease` flag is set.

Comment (BolajiOlajide) [reply testing/tools/upgradetest/tests-util.go]: @DaedalusG Can you re-review when you have some time?

**Review by DaedalusG — APPROVED** [review]

**Pull request merged**